### PR TITLE
Add prepareTableUpdate and its tests

### PR DESCRIPTION
**Pull request by roytman**
Signed-off-by: Alexey Roytman <roytman@il.ibm.com>

### CODE REVIEW

File: pkg/ovsdb/handler.go [hunk 1]
```diff
@@ -107,18 +107,46 @@ func (ch *Handler) Cancel(ctx context.Context, param interface{}) (interface{},
 
 func (ch *Handler) Monitor(ctx context.Context, param ovsjson.CondMonitorParameters) (interface{}, error) {
 	klog.V(5).Infof("Monitor request, parameters %v", param)
-	for tableName, mcr := range param.MonitorCondRequests {
+	ch.mu.Lock()
+	defer ch.mu.Unlock()
+	if ok := ch.createOrDieMonitorsArray(param.JsonValue.(string)); ok != nil {
+		return nil, fmt.Errorf("duplicate json-value")
```
Comment (hunchback) [inline pkg/ovsdb/handler.go:113]: why are you not using error returned by createOrDoeMonoitorsArray?

File: pkg/ovsdb/handler.go [hunk 2]
```diff
@@ -107,18 +107,46 @@ func (ch *Handler) Cancel(ctx context.Context, param interface{}) (interface{},
...
 	return ch.getMonitoredData(param.DatabaseName, param.MonitorCondRequests, false)
 }
 
 func (ch *Handler) MonitorCancel(ctx context.Context, param interface{}) (interface{}, error) {
 	klog.V(5).Infof("MonitorCancel request, parameters %v", param)
-
-	return "{Monitor_cancel}", nil
+	jsonValue, err := common.ParamsToString(param)
+	if err != nil {
+		return map[string]bool{"locked": false}, err
```
Comment (hunchback) [inline pkg/ovsdb/handler.go:136]: just do `return nil, err`

File: pkg/ovsdb/handler.go [hunk 3]
```diff
@@ -198,7 +226,12 @@ func (ch *Handler) Steal(ctx context.Context, param interface{}) (interface{}, e
 
 func (ch *Handler) MonitorCond(ctx context.Context, param ovsjson.CondMonitorParameters) (interface{}, error) {
 	klog.V(5).Infof("MonitorCond request, parameters %v", param)
-
+	ch.mu.Lock()
+	defer ch.mu.Unlock()
+	if ok := ch.createOrDieMonitorsArray(param.JsonValue.(string)); ok != nil {
+		return nil, fmt.Errorf("duplicate json-value")
```
Comment (hunchback) [inline pkg/ovsdb/handler.go:232]: use the error return by `createOrDieMonotirsArray`

File: pkg/ovsdb/handler.go [hunk 4]
```diff
@@ -210,7 +243,12 @@ func (ch *Handler) MonitorCondChange(ctx context.Context, param []interface{}) (
...
+	}
+	ch.jsonValueToUpdateNotificationType[param.JsonValue.(string)] = ovsjson.Update3
 	data, err := ch.getMonitoredData(param.DatabaseName, param.MonitorCondRequests, true)
 	if err != nil {
 		return nil, err
```
Comment (hunchback) [inline pkg/ovsdb/handler.go:254]: you need to cleanup `ch.jsonValueToUpdateNotificationType` if out due to error

File: pkg/ovsdb/monitor_test.go [hunk 5]
```diff
@@ -1,97 +1,242 @@
...
+
+			assert.EqualValuesf(t, ts.expRowUpdate, row, "[%s test] returned wrong row update, expected %#v, got %#v", name, *ts.expRowUpdate, *row)
+		}
+
+	}
+
+}
+
+func TestMonitorResponse(t *testing.T) {
+	/*
```
Comment (hunchback) [inline pkg/ovsdb/monitor_test.go:162]: you really don't want to commit this ... if not in use then delete the content

File: pkg/ovsdb/handler.go (hunk 1 above)
Comment (roytman) [reply pkg/ovsdb/handler.go:113]: fixed

File: pkg/ovsdb/handler.go (hunk 2 above)
Comment (roytman) [reply pkg/ovsdb/handler.go:136]: yeap

File: pkg/ovsdb/handler.go (hunk 3 above)
Comment (roytman) [reply pkg/ovsdb/handler.go:232]: done

File: pkg/ovsdb/handler.go (hunk 4 above)
Comment (roytman) [reply pkg/ovsdb/handler.go:254]: yes, I have to clean up this one and set and clean up `jsonValueToMonitors`.
This part is not done yet, I will work on it today.

File: pkg/ovsdb/monitor_test.go (hunk 5 above)
Comment (roytman) [reply pkg/ovsdb/monitor_test.go:162]: let me think about this test more, I don't want to remove it for now.

**Pull request merged**